### PR TITLE
Remove unnecessary cast() in data.py

### PR DESCRIPTION
**Pull request by junholee6a**
Issue: https://github.com/capitalone/DataProfiler/issues/717

Removing this `cast()` doesn't cause a mypy error.

Simply assigning `dict()` to `options` changes its type from `Union[builtins.dict[Any, Any], None]` to `builtins.dict[Any, Any]`. So casting doesn't do anything here.

### CODE REVIEW

**Comment by taylorfturner** [comment]
hold off merging until #803 is merged. Thx

**Comment by taylorfturner** [comment]
okay ready to go on these, @junholee6a -- update branch and I'll approve and set to auto merge

**Review by taylorfturner — APPROVED** [review]
LGTM thx

**Review by JGSweets — APPROVED** [review]

**Review by ksneab7 — APPROVED** [review]

**Comment by junholee6a** [comment]
Thank you for the review!

**Pull request merged**
